downloads/swan-lake-release-notes/swan-lake-2201.13.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,321 @@
+                currIndex += 1;
+            }
+        }
+        test:assertEquals(currIndex, filteredEmployeesLength);
+    }
+    ```
+
+- Function generation - `@natural:code external` functions
+
+    An `external` function can be annotated with `@natural:code` with the `prompt` field describing the implementation of the function. Note that this is only allowed with projects (and not in single file mode) and the code generated for the function is persisted in the `generated` directory. Since there is no developer in the loop to review and approve the generated code during the build, some restrictions (such as limiting the organizations from which modules can be imported) have been introduced, compared to a Copilot.
+
+    ```ballerina
+    type Employee record {|
+        int id;
+        string name;
+        string department;
+        decimal salary;
+    |};
+
+    function filterEmployees(Employee[] employees, string department, decimal salary) returns Employee[] =
+        @natural:code {
+            prompt: string `Filter employees who belong to the specified department 
+                                and have a salary greater than the given threshold.`
+        } external;
+    ```
+
+    Note that this currently works with the Ballerina Copilot configuration. To generate the configuration, open up the VS Code command palette (`Ctrl` + `Shift` + `P` or `command` + `shift` + `P`), and run the `Configure default WSO2 Model Provider` command to add your configuration to the `Config.toml` file. If not already logged in, log in to the Ballerina Copilot when prompted. Use these values as the `BAL_CODEGEN_URL` and `BAL_CODEGEN_TOKEN` environment variables for compile-time code generation.
+
+### Improvements
+
+#### Improved XML parsing to support XML declarations
+
+Previously, the `xml:fromString()` function did not capture XML declaration(`<?xml version="1.1" encoding="UTF-8"?>`) when converting a string into a xml value. With this update, the XML declaration is now correctly recognized and preserved during parsing.
+
+```ballerina
+import ballerina/io;
+
+public function main() returns error? {
+    string str = string `<?xml version="1.1" encoding="UTF-8"?><A><B>123</B></A>`;
+    xml xmlVal = check xml:fromString(str);
+    io:println(xmlVal); // prints `<?xml version="1.1" encoding="UTF-8"?><A><B>123</B></A>` 
+} 
+```
+
+## Ballerina library updates
+
+### New features
+
+#### `time` package
+
+- Added support for add/subtract specified durations to/from time values.
+
+    ```ballerina
+    import ballerina/io;
+    import ballerina/time;
+
+    public function main() {
+        time:Civil civil = check time:civilFromString("2025-04-13T17:35:30.120Z");
+        civil = check time:civilAddDuration(civil, {years: 1, months: 3, days: 5, hours: 6, minutes: 9, seconds: 1});
+        io:println(check time:civilToString(civil)); // Prints "2026-07-18T23:44:31.120Z"
+
+        time:TimeZone timeZone = check new("Asia/Colombo");
```

Comment:
Better to leave a comment on what is doing here. and for consistency let's add one before line 213 too.

Reply:
PRAM https://github.com/ballerina-platform/ballerina-dev-website/pull/10169